app/policies/authorizer/enforcement/query_interface.rb

```diff
@@ -0,0 +1,97 @@
+module Authorizer
+  # @api private
+  #
+  # This module is the container for authorization enforcement
+  # considerations.
+  #
+  # It does not define how we administer the authorization datastore.
+  # That is a separate (as yet to be described) module; perhaps we
+  # could call it Authorizer::Administration.
+  #
+  # From the application's non-adminsitrative perspective, we want to
+  # avoid asking questions such as `render_edit_button_for(@article)
+  # if user.admin?`.  The non-administrative portion should instead as
+  # `render_edit_button_for(@article) if user.authorized_to?(resource:
```

Comment:
> The non-administrative portion should instead as `render_edit_button_for(@article) if user.authorized_to?@article, action: :edit)`.

Is our long-term goal to get rid of Pundit completely? Because otherwise, this seems like something we can already do right now:

```ruby
render_edit_button_for(@article) if policy(@article).update? # we don't have an `edit?` policy
```

Reply:
In a past application that I developed, I initially considered CanCanCan as the authorization layer.  I then chose Pundit.  But I chose to wrap a module around Pundit.  In doing that, I found it quite easy to test my entire application's authorization model.  Because every controller and view used the same module interface.  And as the application evolved, it became clear that I really didn't need Pundit.

The question of "is our long-term goal to get rid of Pundit" is not something I'm prepared to answer.  But I am prepared to say that I want an insulating layer around the implementation details of Pundit.  And a few reasons for that are:

1) It follows our pattern of wanting a module wrapping gems.
2) Pundit does not allow us to easily answer the question "who all specifically can see this article?"  That is to say the auditability of permissions.

We can and will get a tremendous amount of mileage out of Pundit, but already on the horizon of the feature requests for authorization is the concept that an `:moderator` in one Forem instance might have different authorizations than an `:moderator` for another site.

I also look at the refactor necessary to work towards leveraginig Pundit as is in our views/controllers; that is to say moving away from the `user.admin?` tests in the view.  Given that we need to do that effort, it makes sense to me that we insulate against the implimentation details of Pundit (even with a lightweight wrapper).